packages/debug/src/browser/debug-session-manager.ts

```diff
@@ -424,19 +424,23 @@ export class DebugSessionManager {
             return this.doPostTaskAction(`Could not run the task '${taskName}'.`);
         }
 
-        const code = await this.taskService.getExitCode(taskInfo.taskId);
-        if (code === 0) {
-            return true;
-        } else if (code !== undefined) {
-            return this.doPostTaskAction(`Task '${taskName}' terminated with exit code ${code}.`);
-        } else {
-            const signal = await this.taskService.getTerminateSignal(taskInfo.taskId);
-            if (signal !== undefined) {
-                return this.doPostTaskAction(`Task '${taskName}' terminated by signal ${signal}.`);
+        return Promise.race([this.taskService.getBackgroundTaskEndMatch(taskInfo.taskId), this.taskService.getExitCode(taskInfo.taskId)]).then(async result => {
```

Comment:
it looks tricky to decide on a result depending on the type.

Can we use `Promis.all` instead and have to separate variables?

Also `async/await` is more preferable than using promise chaining.

Reply:
No, i can not use promise.all .
Because if I have a task that depends on a background task, then when the background task matches it end pattern , i want to continue to the root task and not wait until it finished, because it may not finish.

Reply:
> it looks tricky to decide what is a result depending on the type.

fixed.done

Reply:
> Can we use `Promis.all` instead to have separate variables?
> 
> Also `async/await` is more preferable than using promise chaining.

I must use here promise.race since i need to handle the case of depending on background tasks,  and background tasks can send a feedback that the task was "ended" before the task was "exited".